src/lib.rs

```diff
@@ -232,38 +287,67 @@ impl Surface {
 
         Ok(Self {
             surface_impl: Box::new(imple),
+            _marker: PhantomData,
         })
     }
 
-    /// Shows the given buffer with the given width and height on the window corresponding to this
-    /// graphics context. Panics if buffer.len() ≠ width*height. If the size of the buffer does
-    /// not match the size of the window, the buffer is drawn in the upper-left corner of the window.
-    /// It is recommended in most production use cases to have the buffer fill the entire window.
-    /// Use your windowing library to find the size of the window.
-    ///
-    /// The format of the buffer is as follows. There is one u32 in the buffer for each pixel in
-    /// the area to draw. The first entry is the upper-left most pixel. The second is one to the right
-    /// etc. (Row-major top to bottom left to right one u32 per pixel). Within each u32 the highest
-    /// order 8 bits are to be set to 0. The next highest order 8 bits are the red channel, then the
-    /// green channel, and then the blue channel in the lowest-order 8 bits. See the examples for
-    /// one way to build this format using bitwise operations.
+    /// Set the size of the buffer that will be returned by [`Surface::buffer_mut`].
     ///
-    /// --------
-    ///
-    /// Pixel format (u32):
-    ///
-    /// 00000000RRRRRRRRGGGGGGGGBBBBBBBB
-    ///
-    /// 0: Bit is 0
-    /// R: Red channel
-    /// G: Green channel
-    /// B: Blue channel
+    /// If the size of the buffer does not match the size of the window, the buffer is drawn
+    /// in the upper-left corner of the window. It is recommended in most production use cases
+    /// to have the buffer fill the entire window. Use your windowing library to find the size
+    /// of the window.
+    pub fn resize(&mut self, width: u32, height: u32) -> Result<(), SoftBufferError> {
+        self.surface_impl.resize(width, height)
+    }
+
+    /// Return a [`Buffer`] that the next frame should be rendered into. The size must
+    /// be set with [`Surface::resize`] first. The initial contents of the buffer may be zeroed, or
+    /// may contain a previous frame.
+    pub fn buffer_mut(&mut self) -> Result<Buffer, SoftBufferError> {
+        Ok(Buffer {
+            buffer_impl: self.surface_impl.buffer_mut()?,
+            _marker: PhantomData,
+        })
+    }
+}
+
+/// A buffer that can be written to by the CPU and presented to the window.
+///
+/// This derefs to a `[u32]`, which depending on the backend may be a mapping into shared memory
+/// accessible to the display server, so presentation doesn't require any (client-side) copying.
```

Comment:
I think it would be useful to add a `# Platform dependent behavior` documentation which platforms actually support no-copy presentation.